server.js

```diff
@@ -1,35 +1,119 @@
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
+import listEndpoints from "express-list-endpoints";
+import topMusicData from "./data/top-music.json";
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise;
 
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
-const app = express();
+const app = express()
+
+const Song = mongoose.model("song", {
+    id: Number,
+    trackName: String,
+    artistName: String,
+    genre: String,
+    bpm: Number,
+    energy: Number,
+    danceability: Number,
+    loudness: Number,
+    liveness: Number,
+    valence: Number,
+    length: Number,
+    acousticness: Number,
+    speechiness: Number,
+    popularity: Number
+})
+
+if(process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Song.deleteMany()
+    topMusicData.forEach(singleSong => {
+      const newSong = new Song(singleSong)
+      newSong.save()
+    })
+  }
+  seedDatabase()
+}
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors());
-app.use(express.json());
+app.use(cors())
+app.use(express.json())
 
-// Start defining your routes here
+// Routes
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+  const landingPage = {
+    Welcome: "Welcome! This is an open API for Top Music!",
+    Routes: [
+      {
+        "/endpoints": "List of endpoints",
+        "/songs": "Get the Top Music List",
+        "/songs/id/:id": "Get a unique song by id",
+        "/songs/song/:artistName": "Get songs by artistname",
+        "/songs/song?trackName='title of track'": "Get song by trackname",
+        "/songs/song?artistName='name of artist'": "GET songs by artistname",
+      },
+    ],
+  }
+  res.send(landingPage);
+})
+
+app.get("/endpoints", (req, res) => {
+  res.send(listEndpoints(app));
+})
+
+app.get("/songs", async (req, res) => {
+  const allSongs = await Song.find()
+  res.json(allSongs)
+})
+
+//Path params
+app.get("/songs/id/:id", async (req, res) => {
+  //findOne() will retrieve first object with given criteria
+     const songById = await Song.findOne({id: req.params.id})
+  
+     if (songById) {
+       res.status(200).json ({
+         data: songById,
+         success: true,
+       })
+     } else {
+       res.status(404).json({
+         data: "Song not found",
+         success: false,
+       })
```

Comment:
You still managed to implement error handling despite the database issues and it's working great!:D